x/executionlayer/genesis.go

```diff
@@ -39,11 +39,13 @@ func InitGenesis(
 
 	keeper.SetGenesisConf(ctx, data.GenesisConf)
 
-	keeper.SetCandidateBlock(ctx, types.CandidateBlock{
-		Hash:  []byte(types.GenesisBlockHashKey),
-		Bonds: bonds,
-	})
 	keeper.SetEEState(ctx, []byte(types.GenesisBlockHashKey), stateHash)
+
+	candidateBlock := ctx.CandidateBlock()
+	candidateBlock.Hash = []byte(types.GenesisBlockHashKey)
+	candidateBlock.State = stateHash
+	candidateBlock.Bonds = bonds
+	ctx = ctx.WithCandidateBlock(candidateBlock)
```

Comment:
Because it treats candidateBlock as a pointer, it is not necessary to call WithCandidateBlock.

Reply:
😅

Reply:
https://github.com/hdac-io/friday/pull/58/commits/6f1742302db2b5a043fd34ef8f04767522701f87